File: Backend/generic-free-spots-backend/src/main/java/be/PXLResearch/code4belgium/escaperooms/DTO/EscapeRoomDto/EscapeRoomRequest.java
package be.PXLResearch.code4belgium.escaperooms.DTO.EscapeRoomDto;

import be.PXLResearch.code4belgium.enums.City;
import com.fasterxml.jackson.databind.JsonNode;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
public class EscapeRoomRequest {
    private String name;
    private String address;
    private int postalCode;
    private String city;
    private String email;
    private String phoneNumber;
    private String website;
    private JsonNode filterableProperties;
}

File: Frontend/gvpp/src/models/escapeRoomRequest.ts
export class EscapeRoomRequest {
  name: string;
  description: string;
  organizationId: number;
  address: string;
  postalCode: number;
  city: string;
  email: string;
  phoneNumber: string;
  website: string;

  constructor(name: string, description: string, organizationId: number, address: string, postalCode: number, city: string, email: string, phoneNumber: string, website: string) {
    this.name = name;
    this.description = description;
    this.organizationId = organizationId;
    this.address = address;
    this.postalCode = postalCode;
    this.city = city;
    this.phoneNumber = phoneNumber;
    this.website = website;
    this.email = email;
  }
}
